### PR TITLE
feat: Setup Poetry environment

### PR DESCRIPTION
**Pull request by Jimlinz**
This PR adds the following:
- `reset-dev-env.bash` to enable ease of setting up Poetry environment
- `.python-version`
- Update `pyproject.toml` to include project description and authors list

### CODE REVIEW

File: reset-dev-env.bash [hunk 1]
```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+set -o errexit -o noclobber -o nounset -o pipefail
+
+usage() {
+    cat >&2 << 'EOF'
+./reset-dev-env.bash --all
+./reset-dev-env.bash [--delete] [--hooks] [--node] [--python] [--submodule]
```
Comment (l0b0) [inline reset-dev-env.bash:8]: Are `--node` and `--submodule` relevant here?

File: pyproject.toml [hunk 2]
```diff
@@ -45,6 +44,11 @@ pytest = "*"
 pytest-randomly = "*"
 typed-ast = "*"
 
+[tool.poetry.extras]
+cdk = [
```
Comment (l0b0) [inline pyproject.toml:48]: Is this meant to be here? `aws-cdk-lib` is a non-optional production dependency, so it'll be installed whether you specify the extra or not.

Comment (Jimlinz) [reply pyproject.toml]: Was a bit confused about this initially. It appears the `sed` command in `reset-dev-env.bash` looks for `tools.poetry.extras`. Keeping this empty throws an error: `Extra [] is not specified.` A work  around would be to set `cdk= []`, which would get around the problem. The alternative would be to update the bash script. Any preference?

Comment (l0b0) [reply pyproject.toml]: Oh yeah, the reset script is Geostore-specific. You probably want to remove those parts of the script.

File: reset-dev-env.bash [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+./reset-dev-env.bash --all
+./reset-dev-env.bash [--delete] [--hooks] [--python]
+./reset-dev-env.bash --help
+
+`--all` implies `--delete --hooks --node --python --submodule`.
+EOF
+}
+
+arguments="$(getopt --options '' \
+    --longoptions all,delete,help,hooks,node,python,submodule --name "$0" -- "$@")"
```
Comment (l0b0) [inline reset-dev-env.bash:16]: `node` and `submodule` should be removed here as well.

**Comment by l0b0** [comment]
You'll need to resolve the conflict before I can review it.

**Review by l0b0 — APPROVED** [review]

**Pull request merged**